src/NuGet.Licenses/Controllers/LicenseController.cs

```diff
@@ -149,9 +150,66 @@ private ActionResult DisplayLicenseFromFile(string identifier, bool isException)
         private ActionResult DisplayCompositeLicenseExpression(LicenseOperator licenseExpressionRoot, string licenseExpression)
         {
             var meaningfulSegments = _licenseExpressionSegmentator.GetLicenseExpressionSegments(licenseExpressionRoot);
+
             var allSegments = _licenseExpressionSegmentator.SplitFullExpression(licenseExpression, meaningfulSegments);
 
+            var stopwatch = Stopwatch.StartNew();
+            foreach (var segment in allSegments.Where(IsLicenseOrException))
+            {
+                var isValidSegment = false;
+                if (NuGetLicenseData.ExceptionList.TryGetValue(segment.Value, out var exceptionData))
+                {
+                    isValidSegment = true;
+                }
+                else
+                {
+                    try
+                    {
+                        var license = NuGetLicenseExpression.Parse(segment.Value) as NuGetLicense;
+                        if (license != null && license.IsStandardLicense)
+                        {
+                            isValidSegment = true;
+                        }
+                    }
+                    catch (NuGetLicenseExpressionParsingException e)
+                    {
+                        stopwatch.Stop();
+                        _logger.LogError(0, e, "Got exception while attempting to parse license expression {LicenseExpression}", segment.Value);
+                        return InvalidRequest($"Unable to parse the license expression: {segment.Value}");
+                    }
+                }
+
+                if (!isValidSegment)
+                {
+                    stopwatch.Stop();
+                    _logger.LogInformation("Validating composite licenseExpression uses: {elapsedTime} s", stopwatch.Elapsed.TotalSeconds);
```

Comment:
Our team's style is to uppercase the log's parameters. So `{elapsedTime}` should be `{ElapsedTime}`.